src/ember-resource.js

```diff
@@ -819,6 +819,12 @@
 
       Ember.Resource.fetch(this, ajaxOptions)
         .done(function(json) {
+          if (self.get('isDestroying') || self.get('isDestroyed')) {
+            Ember.Resource.sendEvent(self, 'didFail');
+            result.reject();
+            self.fetched().reject();
+            return;
+          }
```

Comment:
Do you think the promise needs to resolve at all? I don't feel strongly about the current code, seemed like the least worst thing to do.